Foundation/URLSession/http/MultiHandle.swift

```diff
@@ -214,12 +214,14 @@ fileprivate extension URLSession._MultiHandle {
         }
         let easyHandle = easyHandles[idx]
         // Find the NSURLError code
-        let errorCode = easyHandle.urlErrorCode(for: easyCode)
-        completedTransfer(forEasyHandle: easyHandle, errorCode: errorCode)
+        let errorInfo = easyHandle.urlErrorCode(for: easyCode).map {
+            URLErrorInfo(code: $0, description: String(cString: CFURLSessionErrorDescription(easyCode.value)))
+        }
+        completedTransfer(forEasyHandle: easyHandle, errorInfo: errorInfo)
     }
     /// Transfer completed.
-    func completedTransfer(forEasyHandle handle: _EasyHandle, errorCode: Int?) {
-        handle.completedTransfer(withErrorCode: errorCode)
+    func completedTransfer(forEasyHandle handle: _EasyHandle, errorInfo: URLErrorInfo?) {
+        handle.completedTransfer(withErrorInfo: errorInfo)
     }
```

Comment:
This should all use Foundation's existing abstraction for errors: `NSError`, instead of a custom struct.

Reply:
We can’t construct an NSError before we pass it around as the code doen’t initially know the URL that is later wrapped in the userInfo.

Reply:
That can be set later by creating another `NSError`, or using the concept of underlying error.

It's important that we don't create several different kinds of abstractions for the same concept in one source base.